File: src/java/nz/co/searchwellington/model/FeedImpl.java
package nz.co.searchwellington.model;

import java.util.Date;
import java.util.Set;



public class FeedImpl extends PublishedResourceImpl implements Feed {
  
    // TODO migrate acceptance options to Enums.
    String acceptancePolicy;
    Date latestItemDate;
    Date lastRead;
    
    public FeedImpl() {
    }
    
    
    public FeedImpl(int id, String name, String url, String description, Website publisher, String acceptancePolicy, Set<Tag> tags) {
        this.id = id;
        this.name = name;
        this.url = url;
        this.description = description;
        this.publisher = publisher;
        this.acceptancePolicy = acceptancePolicy;
        this.tags = tags;
    }
    
    
  
    public String getType() {
        return "F";
     }

   

    public String getAcceptancePolicy() {
        return acceptancePolicy;
    }


    public void setAcceptancePolicy(String acceptancePolicy) {
        this.acceptancePolicy = acceptancePolicy;
    }
    
    
    public Date getLatestItemDate() {
        return latestItemDate;
    }

    public void setLatestItemDate(Date latestPublicationDate) {
        this.latestItemDate = latestPublicationDate;      
    }

    public Date getLastRead() {
        return lastRead;
    }

    public void setLastRead(Date lastRead) {
        this.lastRead = lastRead;
    }
    
    public boolean isScreenScraped() {
    	return url.startsWith("http://brownbag.wellington.gen.nz/");    	
    }
    
        
}

File: src/main/scala/nz/co/searchwellington/model/FeedImpl.scala
package nz.co.searchwellington.model

import java.util.Date

import org.joda.time.format.ISODateTimeFormat

case class FeedImpl(override var id: Int = 0,
                    override var `type`: String = "",
                    override var title: Option[String] = None,
                    override var description: Option[String] = None,
                    override var page: Option[String] = None,
                    override var http_status: Int = 0,
                    override var date2: Option[Date] = None,
                    override var last_scanned2: Option[Date] = None,
                    override var last_changed2: Option[Date] = None,
                    override var live_time2: Option[Date] = None,
                    override var embargoed_until2: Option[Date] = None,
                    override var held2: Boolean = true,
                    override var url_words: Option[String] = None,
                    override var geocode: Option[Int] = None,
                    override var owner: Option[Int] = None,
                    var acceptance: String = "IGNORE",
                    var latestItemDate2: Option[Date] = null,
                    var lastRead2: Option[Date] = null, // TODO
                    var whakaoko_id: String = null,
                    override var publisher: Option[Int] = None) extends PublishedResource with Feed {

  override def getType: String = "F"

  override def getAcceptancePolicy: String = acceptance

  override def setAcceptancePolicy(acceptancePolicy: String): Unit = this.acceptance = acceptancePolicy

  override def getLatestItemDate: Date = latestItemDate2.getOrElse(null)

  override def setLatestItemDate(latestPublicationDate: Date): Unit = {}

  override def getLastRead: Date = lastRead2.getOrElse(null)

  override def setLastRead(lastRead: Date): Unit = {}

  def isScreenScraped: Boolean = page.map(p => p.startsWith("http://brownbag.wellington.gen.nz/")).getOrElse(false)

  override def getWhakaokoId: String = whakaoko_id

  override def setWhakaokoId(whakaokoId: String): Unit = this.whakaoko_id = whakaokoId

}
